pyfr/plugins/ascent.py

```diff
@@ -0,0 +1,415 @@
+from ast import literal_eval
+from collections import defaultdict
+from ctypes import (c_char_p, c_double, c_float, c_long, c_longlong, c_void_p,
+                    RTLD_GLOBAL)
+import re
+
+import numpy as np
+
+from pyfr.ctypesutil import LibWrapper
+from pyfr.mpiutil import get_comm_rank_root
+from pyfr.nputil import npeval
+from pyfr.plugins.base import BasePlugin, RegionMixin
+from pyfr.shapes import BaseShape
+from pyfr.util import file_path_gen, subclass_where
+from pyfr.writers.vtk import BaseShapeSubDiv
+
+
+class AscentError(Exception): pass
+class ConduitError(Exception): pass
+
+
+class ConduitWrappers(LibWrapper):
+    _libname = 'conduit'
+    _mode = RTLD_GLOBAL
+
+    # Functions
+    _functions = [
+        (c_void_p, 'conduit_node_append', c_void_p),
+        (c_void_p, 'conduit_node_create', c_void_p),
+        (None, 'conduit_node_destroy', c_void_p),
+        (c_void_p, 'conduit_node_fetch', c_void_p, c_char_p),
+        (c_void_p, 'conduit_node_set_path_char8_str', c_void_p, c_char_p,
+         c_char_p),
+        (c_void_p, 'conduit_node_set_path_float32', c_void_p, c_char_p,
+         c_float),
+        (c_void_p, 'conduit_node_set_path_float32_ptr', c_void_p, c_char_p,
+         c_void_p, c_longlong),
+        (c_void_p, 'conduit_node_set_path_float64', c_void_p, c_char_p, 
+         c_double),
+        (c_void_p, 'conduit_node_set_path_float64_ptr', c_void_p, c_char_p,
+         c_void_p, c_longlong),
+        (c_void_p, 'conduit_node_set_path_int64', c_void_p, c_char_p,
+         c_longlong),
+        (c_void_p, 'conduit_node_set_path_int64_ptr', c_void_p, c_char_p,
+         c_void_p, c_longlong),
+        (c_void_p, 'conduit_node_set_path_node', c_void_p, c_char_p, c_void_p),
+        (c_void_p, 'conduit_node_print', c_void_p),
+    ]
+
+    def _errcheck(self, status, fn, args):
+        if not status:
+            raise ConduitError
+
+
+class ConduitNode:
+    def __init__(self, lib, ptr=None, child=False):
+        self.lib = lib
+        self.child = child
+        self._as_parameter_ = ptr or self.lib.conduit_node_create(None)
+
+    def __del__(self):
+        if not self.child:
+            self.lib.conduit_node_destroy(self)
+
+    def __setitem__(self, key, value):
+        key = key.encode()
+        match value:
+            case str():
+                self.lib.conduit_node_set_path_char8_str(self, key, 
+                                                         value.encode())
+            case ConduitNode():
+                self.lib.conduit_node_set_path_node(self, key, value)
+            case int():
+                self.lib.conduit_node_set_path_int64(self, key, value)
+            case float():
+                self.lib.conduit_node_set_path_float32(self, key, value)
+            case (np.ndarray() | np.generic() | list()):
```

Comment:
Remove the list support as it is broken.

Reply:
It is not required.  The calling code just needs to convert it's list to a suitably typed NumPy array before assigning.  The issue with lists is that blindly converting them to a NumPy array can result in weirdness `[-1, 0]` and `[-1, 0.0]` end up as different NumPy arrays (one int, and one float).  When calling down to C functions one can not be cavalier about data types.

Reply:
I don't really know a good solution to this. The lists can be both floats and ints depending on what they are used for, and there are too many options to list them all and check. 

This seemed like the best option by, falling back to Ascents error handling. I'm open to how this can be handled better, currently the information is read form the cfg via getliteral, and from playing you don't seem to be able to do either:
```
x = ast.literal_eval('numpy.array([1,2,3])')
y = numpy.safe_eval('numpy.array([1,2,3])')
```